FEATURE: Make neos login-screen logo configurable

johannessteu opened the pull request:
With this change it is possible to easily change the logo rendered above the login form. To do so you can adjust the `partialRootPathPattern` for the LoginController in your `Views.yaml` and include your custom logo partial

kitsunet commented:
Uhm, do we want this? I mean this is a Neos, I don't see why you should be able to change that...

johannessteu commented:
Sure, something to discuss about. I think as we don't brand the backend itself with some neos logos etc an integrator should also be abel to easily change the logo on the login screen. For me you don't login into "Neos" but the site you are using. Neos is the system you do use, but is nothing you log in to actually. So i think a editor would be more confident with the system if the login screen renders "his logo".
 
I added a small neos logo to the copyright claim so we would have it placed at least once.

So this is how it could look like if you implement your own stuff. Neos demo vs custom site:
![screen shot 2016-05-28 at 11 19 10](https://cloud.githubusercontent.com/assets/769789/15626511/98ad7ba2-24c6-11e6-99a2-eedcdab2505f.png)
![screen shot 2016-05-28 at 11 16 39](https://cloud.githubusercontent.com/assets/769789/15626512/98dc42de-24c6-11e6-9bac-6661050686c7.png)

kitsunet commented:
right, would work for me... 👍

gerhard-boden commented on TYPO3.Neos/Documentation/HowTos/CustomLoginScreen.rst:
```diff
@@ -24,6 +24,18 @@ Or alternatively add a custom stylesheet::
+      partialRootPathPattern: resource://Neos.Demo/Private/Partials
+
+
+With this example Neos would assume a logo in ``Packages/Sites/Neos.Demo/Resources/Private/Partials/Login/Logo.hml``
```
``Logo.html`` instead of  ``Logo.hml``

gerhard-boden replied:
(optional) Maybe we could also mention that the height of `.neos[style] .neos-background-image-clone` should be adapted when the new logo has a different height?

aertmann replied:
good point, maybe see if there's a way to avoid that being necessary?

gerhard-boden commented on TYPO3.Neos/Resources/Private/Templates/Login/Index.html:
```diff
@@ -120,6 +108,18 @@
 		</div>
 		<div id="neos-login-footer">
 			<p>
```
Is it necessary to wrap the logo in a paragraph?

gerhard-boden commented:
nice addition, can be really useful when branding the login is necessary.

Left some nitpicks plus i noticed that the login box is slighty touching the logo in the footer
![image](https://cloud.githubusercontent.com/assets/10533739/15634805/b3217ab2-25cd-11e6-941f-7eba113bfe71.png)

aertmann commented:
I don't mind making this possible, as long as there the branding is still present and not a recommended thing to change. After all Neos is just the tool used to build lots of different things, if those things thrive, Neos will thrive.

But while working on this we should create an actual SVG file with the logo instead of putting it into a template, that's makes no sense. Instead we need to provide an optimized SVG that can be reused and cached, then that can be used for the default logo and below.

This also means we can get rid of putting that into a partial and simplify it by simply allowing you to choose an image, similar to the background image. Only thing we might want to do is to allow setting the width and height as well since that's needed for SVGs.

Also another thing that could make sense is to allow an absolute URL in addition to the local resource pointer for the login logo and the login background image.

aertmann commented:
one issue we might face is that the logo over the copyright claim will clash when using a custom background image, so we probably have to change it to the white outline one in that case and add some text-shadow like with the text

btw. regarding not branding the backend itself was actually justified by at least having the login screen branded

johannessteu commented:
Sorry for the delay ...

I think i tackeled the comments.

Regarding
> @gerhard-boden: Left some nitpicks plus i noticed that the login box is slighty touching the logo in the footer
Yeah, but on starting on a browser height of  600 or something. Should we add some Breakpoints to the flexbox stuff here? There are also some issues if you go for 600px width and smaller. But we might want to tackle this in a follow up?

> @aertmann This also means we can get rid of putting that into a partial and simplify it by simply allowing you to choose an image, similar to the background image. Only thing we might want to do is to allow setting the width and height as well since that's needed for SVGs.

I don't agree on this. As you said if you add a svg you want to add more configuration options, you might also want to add a class name or something to style something more like colours. Our Neos logo is also used with different colours for instance. It does not feel that straight forward. I'd stay for a partial option. It would also enable the users to add a claim or something below their logo.


Sooo, good to go?

johannessteu commented:
Btw the style ci stuff is nothing i touched here. But i think it is also fixed in another PR but i did not rebase this one yet.

gerhard-boden commented:
Hi,
no need to be sorry, happy to see you back :)

> Yeah, but on starting on a browser height of 600 or something. Should we add some Breakpoints to the flexbox stuff here? There are also some issues if you go for 600px width and smaller. But we might want to tackle this in a follow up?

I don't think that the flexbox alignment is the real reason for problems with smaller widths, but more likely hardcoded values like this https://github.com/neos/neos-development-collection/blob/master/TYPO3.Neos/Resources/Private/Styles/Login.scss#L53 
The problem with smaller height comes from using `position: absolute` for the footer.

The login screen could really need some love for small devices, but since there is currently no real mobile support  for the backend that's not too important for the moment I guess. 
But the "footerlogo is touching login box" issue occurs on my notebook with a resolution of 1366x768px. Not too uncommon for non HD notebook displays. So I would suggest to at least fix this issue. According to these stats even 35% use this resolution:
http://www.w3schools.com/browsers/browsers_display.asp


![image](https://cloud.githubusercontent.com/assets/10533739/15910740/a2cf6d8e-2dcc-11e6-8d44-ee415c118274.png)
![image](https://cloud.githubusercontent.com/assets/10533739/15910745/a94a8900-2dcc-11e6-83bb-aea57190d963.png)


I'll repost the  blurry clone overlay as an optional task, since it got folded in the comments after the last commit. 

- [ ] footerlogo should not touch loginbox on 1366x768.
- [ ] optional: Maybe we could also mention that the height of `.neos[style] .neos-background-image-clone` should be adapted when the new logo has a different height? Or even find a way to adjust the height of this element automatically like @aertmann suggested.

gerhard-boden commented:
Had a talk with @johannessteu today. I'm going to remedy (at least some of the) shortcomings of the login screen CSS. This should at least fix the logo bug mentioned above.

johannessteu commented:
Yeah, i'll mark this as WIP for now and rebase this PR on @gerhard-boden PR for the css stuff.

gerhard-boden commented:
done: 
https://github.com/neos/neos-development-collection/pull/566

gerhard-boden commented on TYPO3.Neos/Documentation/HowTos/CustomLoginScreen.rst:
```diff
@@ -34,7 +34,8 @@ inside a ``Views.yaml`` to include your custom logo partial::
 
-With this example Neos would assume a logo in ``Packages/Sites/Neos.Demo/Resources/Private/Partials/Login/Logo.hml``
+With this example Neos would assume a logo in ``Packages/Sites/Neos.Demo/Resources/Private/Partials/Login/Logo.html``
+Note: Depending on your logo you might want to adjust the height for the css class ``.neos[style] .neos-background-image-clone``
```
This change is obsolete, and the seperate layout element is gone: https://github.com/neos/neos-development-collection/pull/566/files#diff-2d322c760eb62cf1d3f4102279ec89baL24

kitsunet commented:
What's the state of this now?

Also removing the logo in the footer invalidates that pro argument of making the big logo changeable. It means no Neos CI will remain in that case and I am not sure I like that. @robertlemke @fheinze ?

johannessteu commented:
@kitsunet  current state/idea is to add the logo the footer and this is not configurable. The "big logo" above the login form should be configurable via a Partiall @gerhard-boden worked on some css issues in #566 . I'd rebase this on his changes so we don't have the issues with the footer logo is touching the login box and a broken blur effect.

johannessteu commented:
Gonna rebase this today as #567 is merged

johannessteu commented:
Rebased the PR. Also added the inline svg in the footer so we can be safe that there is a logo present. If we would use a partial as it was earlier the partial path would also be overwritten if you change the logo ;)

aertmann commented:
@kitsunet: hey, why was this merged without any votes?

It should have been rebased on top of #567, but that hasn't been upmerged so now there will be a painful upmerge.

Also had some feedback to discuss about the change. But was waiting until it was rebased on top of the other change.

I think we need to revert it and make sure a new PR is based on #567.

What I don't really like is the inlined svg.

Also I do find it a little odd way of configuring a new logo, I'd prefer a configuration over a partial and Views.yaml logic. Cause it is already possible to override the view using Views.yaml to achieve it. And if you have to provide custom css anyway, I'd kinda prefer not encouraging people to change it since it's not that straight forward. However it might be that #567 makes it that you don't need to provide custom css.

kitsunet commented:
My bad, @johannessteu said it was upmerged so I thought that was fine. Ok for me to revert, sorry for the mess. Seemed to me that everything was fine after this rebase.

johannessteu commented:
Hey, sorry for that. I just saw that #567 is merged. I had in mind that that was also agains master and didn't thought about upmerge.

aertmann commented:
Okay it happened, the login screen is completely broken now after @kdambekalns upmerged the changes from #567. I'm gonna revert this one and revert the merge conflicts changes he did.

![image](https://cloud.githubusercontent.com/assets/903567/16699861/9871025c-450b-11e6-9d82-e5b39c0e2d30.png)

aertmann commented:
Hmm can't revert it through the UI because of the merge conflict changes. Will do manually.

aertmann commented:
okay that was a bit difficult, but managed..

@johannessteu: you might have some issues when pushing a new PR since the changes have been reverted, https://git-scm.com/blog/2010/03/02/undoing-merges.html might be helpful

johannessteu commented:
So just to get it right, this PR broke the login screen @aertmann ? This PR didn't changed anything regarding css or css classes or something. So i guess #567 causes the trouble?

I'll create a new clean PR based on latest master for this change.

aertmann commented:
@johannessteu: no it didn't so to speak, but it was merged without any votes and when #567 was upmerged improperly it clashed.. it's really a combination of things here that went wrong in this case

so thanks for creating a new PR, you didn't do anything wrong btw.

johannessteu commented:
Yeah, was just wondering @aertmann . Just a idea for this change since you are not good with the inline svg and i don't like the idea of a configurable logo file because if you go for svg you need more stuff like width and height. What about a configurable logo template path? The default would render a generic neos logo partial, the same we would render in the footer. But as an editor you would still be able to add whatever you want to in that partial. For example a logo + a claim or something like that
